# UN Data Project - GDP and Internet Usage

In [157]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

Using the pandas read_csv() method, read the GDP dataset into your notebook as a DataFrame called gdp_df. Take a look at the first few and last few rows to familiarize yourself with what is contained in this dataset.

In [158]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')
print(gdp_df.head())
print(gdp_df.tail())

  Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2020  1970.560169              NaN
1     Afghanistan  2019  2065.036235              NaN
2     Afghanistan  2018  2033.804389              NaN
3     Afghanistan  2017  2058.400221              NaN
4     Afghanistan  2016  2057.067978              NaN
     Country or Area  Year        Value  Value Footnotes
7171        Zimbabwe  1994  2603.927369              NaN
7172        Zimbabwe  1993  2420.092791              NaN
7173        Zimbabwe  1992  2437.152908              NaN
7174        Zimbabwe  1991  2733.696626              NaN
7175        Zimbabwe  1990  2652.129209              NaN


<b>How many rows and columns are in gdp_df? What are the data types of each column?</b>

In [159]:
print(gdp_df.dtypes)

Country or Area     object
Year                 int64
Value              float64
Value Footnotes    float64
dtype: object


There are 7,175 rows and 4 columns. The column data types can be seen in the above cell.

<b>Drop the Value Footnotes column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.</b>

In [160]:
gdp_df.drop('Value Footnotes', axis=1, inplace=True)
gdp_df.rename(columns={'Country or Area': 'Country', 'Value': 'GDP_Per_Capita'}, inplace=True)
print(gdp_df.head())

       Country  Year  GDP_Per_Capita
0  Afghanistan  2020     1970.560169
1  Afghanistan  2019     2065.036235
2  Afghanistan  2018     2033.804389
3  Afghanistan  2017     2058.400221
4  Afghanistan  2016     2057.067978


<b>How many countries have data for all years? Which countries are missing many years of data? Look at the number of observations per year. What do you notice?</b>

In [161]:
min_year = gdp_df['Year'].min()
max_year = gdp_df['Year'].max()

# Using a lambda function to filter for countries with data for every year
countries_with_data_df = gdp_df.groupby('Country').filter(lambda x: len(x) == (max_year - min_year + 1)).nunique()
print('Number of countries with data all years: ', countries_with_data_df['Country'])

# We are defining many years as 10 or more years of data
countries_missing_data_df = gdp_df.groupby('Country').filter(lambda x: len(x) < (max_year - min_year - 10)).nunique()
print('Number of countries missing 10+ years:   ', countries_missing_data_df['Country'])

# Getting the number of observations per year
observations_per_year_df = gdp_df.groupby('Year').count()
print(observations_per_year_df['Country'])


Number of countries with data all years:  205
Number of countries missing 10+ years:    8
Year
1990    206
1991    207
1992    208
1993    209
1994    211
1995    222
1996    222
1997    226
1998    226
1999    227
2000    233
2001    234
2002    235
2003    235
2004    236
2005    236
2006    237
2007    237
2008    238
2009    239
2010    239
2011    240
2012    240
2013    242
2014    242
2015    242
2016    242
2017    242
2018    242
2019    241
2020    240
Name: Country, dtype: int64


205 countries have data reported for every year. 8 countries have missed 10 or more years. The total number of observations per year has steadily increased until 2019 and 2020 where 1 and 2 countries, respectively, did not report data.